cedar-policy-core/src/ast/policy.rs

```diff
@@ -421,23 +593,35 @@ pub struct Policy {
     /// The constructor `new` is only visible in this module,
     /// so it is the responsibility of callers to maintain
     values: HashMap<SlotId, EntityUID>,
+    /// All of the generalized slots in `template` MUST
+    /// be bound by `generalized values`.
+    /// The SlotId for generalized_values should be disjoint from
+    /// values as well
+    generalized_values: HashMap<SlotId, RestrictedExpr>,
```

Comment:
I think we could say, more strongly, not just that the SlotIds need to be disjoint, but that all SlotIds other than `?principal` and `?resource` need to be in `generalized_values`.